core/src/main/scala/com/softwaremill/react/kafka/KafkaGraphStageSource.scala

```diff
@@ -74,29 +74,25 @@ class KafkaGraphStageSource[K, V](
               else buffer = None
             }
             else scheduleOnce(TimerPollKey, pollRetryDelayMs)
-          case Failure(ex) =>
-            close()
-            fail(out, ex)
+          case Failure(ex) => fail(out, ex)
         }
       }
 
       def performManualCommit(): Unit = {
-        logger.debug(s"Flushing offsets to commit. Registered offsets: $partitionOffsetMap vs $committedOffsetMap")
-        val offsetMapToFlush = partitionOffsetMap.diff(committedOffsetMap)
+        val partitionOffsetMapCopy = partitionOffsetMap.copy()
```

Comment:
Looks dirty, but it is the simplest solution I found with current shared mutable variable design.